diff --git a/app/policies/custom_emoji_policy.rb b/app/policies/custom_emoji_policy.rb
index a8c3cbc73..0d73e9a15 100644
--- a/app/policies/custom_emoji_policy.rb
+++ b/app/policies/custom_emoji_policy.rb
@@ -6,15 +6,15 @@ class CustomEmojiPolicy < ApplicationPolicy
   end
 
   def create?
-    admin?
+    staff?
   end
 
   def update?
-    admin?
+    staff?
   end
 
   def copy?
-    admin?
+    staff?
   end
 
   def enable?
@@ -26,6 +26,6 @@ class CustomEmojiPolicy < ApplicationPolicy
   end
 
   def destroy?
-    admin?
+    staff?
   end
 end
